tests/unit/core/reducers/test_versions.js

```diff
@@ -342,6 +342,20 @@ describe(__filename, () => {
         ).toEqual(createInternalVersion(version));
       });
 
+      it('handles invalid shelves', () => {
+        const state = versionsReducer(
+          undefined,
+          loadHomeAddons({
+            collections: [],
+            shelves: {
+              featuredExtensions: null,
+            },
+          }),
+        );
+
+        expect(getVersionById({ state, id: versionId })).toEqual(null);
```

Comment:
Because `versionId` is irrelevant to the test, perhaps this should just check that the state after calling `loadHomeAddons` is still `initialState`?